### DISCUSSION TITLE
How to use Performance Data Collection?

### DISCUSSION THREAD

**Discussion by leomeza1** [nasa/cFS#666] [General]
I see in the cfe-userguide.pdf, section 1.6.14 "Performance Data Collection", there is some description of starting and stopping the data collection. But, the description doesn't really go into much details about what is a "trigger mask" or how to set it to get the desired results (e.g. data collection of a specific PERF ID). Does anyone know where I might be able to find an example of this use case?

**Comment by dmccomas** [comment]
OpenSatKit, https://github.com/OpenSatKit/OpenSatKit has a performance
monitor demo script at
cosmos/config/targets/CFE_ES/procedures/demo_perf_mon.rb that steps through
the process. There’s also a video at

https://youtu.be/yGQDtJgFEXM

**Comment by leomeza1** [comment]
@dmccomas thanks! I wasn't able to get OpenSatKit/Cosmos working on my PC, but from the demo_perf_mon.rb script I was able to understand which commands to send my cFS target, and was able to get the cFS performance monitor data (a binary file) based on my applications PERF ID. Now, I'm stuck on trying to get that perfutils-java tool working, but that's another topic, and I've created a new discussion in the OpenSatKit github: 
https://github.com/OpenSatKit/OpenSatKit/discussions/109#discussion-5068835
